newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -57,254 +57,113 @@ type RootMenuTemplate =
       submenu: Array<MenuItemTemplate>,
     |};
 
+// Custom hook to run function only once, regardless of prop changes
+const useMountEffect = func => React.useEffect(func, []);
```

Comment:
I don't know if this is the correct approach or not, but let's go with this for now. I guess the linter is not able to find potential deps while using `useMountEffect`. I believe there's nothing we can do about the functions passed as props at the moment. When the `MainFrame` will get converted into a functional component, then we might wrap these functions with `useCallback` in `MainFrame` itself, which should be the proper way.

Reply:
I did think about cleaning up the listeners but haven't tried it yet. But now that you mention it, I just thought of a possibly cleaner approach - a custom `useEffect` based hook that registers a function for a certain event, and in cleanup removes that function. Seems intuitive, should work well with `useCallback` in the MainFrame, and will be much cleaner in code. I'll try to implement this in a while.

Reply:
I have an alternative implementation ready [here](https://github.com/NilayMajorwar/GDevelop/blob/emm-refactor/ipc-hook/newIDE/app/src/MainFrame/ElectronMainMenu.js) that uses a custom hook for registering and cleaning up IPC event listeners. Both of them work well, but I'm not sure which one is better.

Reply:
It's a very interesting case. Basically, the linter warned us of something that is odd:
we want to use useEffect with an empty array (to run it only at the component mounting), but then the linter says "hey you're using some props, what are you doing dude??".

Basically, this put the stress on the fact that the existing implementation had a bug! There was no unregistering of the event listener.
It was no big deal because there was no need to unregister the listeners... but still it was not "perfect".

React Hooks are in a way forcing us to write correct, proper code that is cleaning the stuff after us. :) 

> I have an alternative implementation ready here that uses a custom hook for registering and cleaning up IPC event listeners. 

I think your alternative implementation is the proper one to use. It's properly unregistering the callbacks.
This means that previously, if someone wanted to play with the menu, like removing the component then re-render it later (I don't know why, but for example if you wanted to hide the menu for some reason) then... there would be a bug because of the listeners.
The alternative implementation seems, from what I've seen, to do the proper job so that the component is now safe! 👍

Reply:
Thanks both for your nice collaboration on this. These are tricky but interesting parts of React. Understanding them well allow us to make robust code that works in 100% of the cases.